curriculum/challenges/english/25-front-end-development/lab-project-idea-board/67051431a73f1ca25d9a6f25.md

```diff
@@ -0,0 +1,334 @@
+    "An automation project allowing users to lock, unlock windows automatically based on weather conditions.",
+  status: { description: "Pending Execution" },
+});
+```
+
+Calling `updateProjectStatus(projectStatus.SUCCESS)` on `new ProjectIdea("Fitness Tracker App", "An app that tracks user workouts, diet, and sleep patterns.")` should update the status to `{ description: 'Executed Successfully' }`.
+
+```js
+let mobileApp = new ProjectIdea(
+  "Fitness Tracker App",
+  "An app that tracks user workouts, diet, and sleep patterns."
+);
+
+// Update the project status to SUCCESS
+mobileApp.updateProjectStatus(projectStatus.SUCCESS);
+assert.deepEqual(mobileApp, {
+  title: "Fitness Tracker App",
+  description: "An app that tracks user workouts, diet, and sleep patterns.",
+  status: { description: "Executed Successfully" },
+});
+```
+
+Calling `updateProjectStatus(projectStatus.FAILURE)` on `new ProjectIdea("Breakfast Chef Robot", "A robot that can follow a given list of instructions and prepare breakfast for the user and let them know through their phone.")` should update the status to `{ description: 'Execution Failed' }`.
+
+```js
+let breakfastChef = new ProjectIdea(
+  "Breakfast Chef Robot",
+  "A robot that can follow a given list of instructions and prepare breakfast for the user and let them know through their phone."
+);
+
+// Update the project status to FAILURE
+breakfastChef.updateProjectStatus(projectStatus.FAILURE);
+
+assert.deepEqual(breakfastChef, {
+  title: "Breakfast Chef Robot",
+  description:
+    "A robot that can follow a given list of instructions and prepare breakfast for the user and let them know through their phone.",
+  status: { description: "Execution Failed" },
+});
+```
+
+Calling `updateProjectStatus(projectStatus.SUCCESS)` on `new ProjectIdea("Online Used Video Games Store", "An online platform where users can buy second hand physical copies of video games from other users.")` should update the status to `{ description: 'Executed Successfully' }`.
+
+```js
+let usedVideoGamesStore = new ProjectIdea(
+  "Online Used Video Games Store",
+  "An online platform where users can buy second hand physical copies of video games from other users."
+);
+
+// Update the project status set to SUCCESS
+usedVideoGamesStore.updateProjectStatus(projectStatus.SUCCESS);
+
+assert.deepEqual(usedVideoGamesStore, {
+  title: "Online Used Video Games Store",
+  description:
+    "An online platform where users can buy second hand physical copies of video games from other users.",
+  status: { description: "Executed Successfully" },
+});
+```
+
+When you pin `new ProjectIdea("Smart Home System", "An integrated system to control lighting, temperature, and security devices remotely.")` to `new ProjectIdeaBoard("Tech Projects Board")`, your project board should show `Tech Projects Board has 1 idea(s)\nSmart Home System (Pending Execution) - An integrated system to control lighting, temperature, and security devices remotely.`
```

Comment:
This is still confusing. When you use `pin`, you push a `ProjectIdea` to `ProjectIdeaBoard.ideas`. That's it.

It shouldn't be correlated to the result of calling `formatToString` after pinning an object.

`formatToString` should be tested in a separate test, setting the `ideas` array to a known content and checking the output.

Here I would simply test that calling `pin` adds the object to the `ideas` array. The same holds for `unpin`.

Reply:
Now, I have updated the hints and tested the inputs againtst the `ideas` array.

```
You should be able to pin a `ProjectIdea` object to your `ProjectIdeaBoard` using the `pin` method.
```
```
You should be able to unpin a `ProjectIdea` object to your `ProjectIdeaBoard` using the `unpin` method.
```